p2p/host/resource-manager/README.md

```diff
@@ -28,9 +28,21 @@ scalingLimits := rcmgr.DefaultLimits
 // Add limits around included libp2p protocols
 libp2p.SetDefaultServiceLimits(&scalingLimits)
 
-// Turn the scaling limits into a static set of limits using `.AutoScale`. This
+// Turn the scaling limits into a reified set of limits using `.AutoScale`. This
 // scales the limits proportional to your system memory.
-limits := scalingLimits.AutoScale()
+scaledDefaultLimits := scalingLimits.AutoScale()
+
+// Tweak certain settings
+cfg := rcmgr.LimitConfig{
+  System: &rcmgr.ResourceLimits{
+    // Allow unlimited outbound streams
+    StreamsOutbound: rcmgr.Unlimited,
+  },
+  // Everything else is default. The exact values will come from `scaledDefaultLimits` above.
+}
+
+// Create our limits by using our cfg and replacing the default values with values from `scaledDefaultLimits`
+limits := cfg.Reify(scaledDefaultLimits)
```

Comment:
A couple of thoughts:
1. Is the word "reify" intuitive enough?  This may certainly just be me.  I didn't know what the word meant, and haven't seen it used in these kind of contexts.  (But if this is a convention in golang or libraries then great.  I just wanted to make sure we're not being too clever here.)
2. The comment sentence reads weird to me.  Is that really what is happening?  I would expect that we have some scaled/computed defaults as a result of the AutoScale call and then we are overriding some of those values by the user-supplied overrides.

Reply:
I guess you're hitting on this in the overall PR description.  I think I need to come at this with a clear head.  Feel free to disregard this for now.

Reply:
Happy to use different terminology. `PartialConfig` vs `ConcreteConfig`? Reification is familiar to me, but if it feels weird for other folks we should change.

Reply:
@MarcoPolo that terminology sounds better.

Reply:
What about `Build`? `Materialize` has a specific meaning in databases that I don’t want to invoke.

Reply:
Changed:
`LimitConfig` => `PartialLimitConfig`
`ReifiedLimitConfig` => `ConcreteLimitConfig`
`.Reify(` => `.Build(`